lib/dal/src/attribute/value.rs

```diff
@@ -1944,4 +1946,72 @@ impl AttributeValue {
             None => None,
         })
     }
+
+    pub async fn get_parent_av_id_for_ordered_child(
+        ctx: &DalContext,
+        id: AttributeValueId,
+    ) -> AttributeValueResult<Option<AttributeValueId>> {
+        let workspace_snapshot = ctx.workspace_snapshot()?;
+
+        let ordering_node_id = match workspace_snapshot
+            .incoming_sources_for_edge_weight_kind(id, EdgeWeightKindDiscriminants::Ordinal)
+            .await?
+            .first()
+            .copied()
+        {
+            Some(ordering_idx) => workspace_snapshot.get_node_weight(ordering_idx).await?.id(),
+            None => return Ok(None),
+        };
+
+        let parent_av_id = if let Some(parent_av_idx) = workspace_snapshot
+            .incoming_sources_for_edge_weight_kind(
+                ordering_node_id,
+                EdgeWeightKindDiscriminants::Ordering,
+            )
+            .await?
+            .first()
+            .copied()
+        {
+            let parent_av_id: AttributeValueId = workspace_snapshot
+                .get_node_weight(parent_av_idx)
+                .await?
+                .id()
+                .into();
+
+            let prop_id = AttributeValue::prop(ctx, parent_av_id).await?;
+
+            let parent_prop = Prop::get_by_id(ctx, prop_id).await?;
+
+            if !vec![PropKind::Map, PropKind::Array].contains(&parent_prop.kind) {
```

Comment:
I think you might be able to use a static array here since those enum variants are stack-allocated? This is a nit, so feel free to ignore.

Reply:
good catch! The linter wouldn't let us merge without fixing  this too. We changed it